contracts/amm/SirenExchange.sol

```diff
@@ -23,53 +23,77 @@ import "@uniswap/lib/contracts/libraries/TransferHelper.sol";
 contract SirenExchange is ERC1155Holder {
     IERC1155 public immutable erc1155Controller;
 
-    constructor(IERC1155 erc1155Controller_) public {
-        erc1155Controller = erc1155Controller_;
+    uint256 private constant _NOT_ENTERED = 1;
```

Comment:
Instead of adding this code, you should just inherit from the [ReentrancyGuard](https://docs.openzeppelin.com/contracts/3.x/api/utils#ReentrancyGuard) contract, that will make the SirenExchange logic cleaner.

The reason I copy + pasted in my PR was because the new MinterAmm.sol logic contract would be used to upgrade existing MinterAmm.sol contracts. I wasn't sure that inheriting would work with existing contracts because the layout of the contract's state variables might have been been screwed up. You can read more about this here: https://docs.openzeppelin.com/upgrades-plugins/1.x/proxies#storage-collisions-between-implementation-versions

So it should be

`contract SirenExchange is ERC1155Holder, ReentrancyGuard {`

Reply:
Gotcha I can add that in